### PR TITLE
Allow users to request account deletion via settings page

### PR DESCRIPTION
**Pull request by luchaos**
Screenshot 2020-11-14 at 16 31 11
Screenshot 2020-11-14 at 16 31 19

In regards to checking for the Deleted timestamp on users only the check in getAccountDetails has been added. 
Some additional checks in other queries might be appropriate down the line.
This PR is meant to deal with the minimum requirements for account deletion for now.

### CODE REVIEW

File: public/controlpanel.php [hunk 1]
```diff
@@ -664,6 +664,29 @@ function onUploadImageComplete(data) {
         </form>
         </div>
         */ ?>
+        <div class='component'>
+            <h3>Delete Account</h3>
+            <p>
+                Your account will be deleted after a 14 days grace period.<br>
+                You will be able to cancel your request within that period.<br>
```
Comment (meleu) [inline public/controlpanel.php:671]: trying to make it shorter, I suggest replacing these two sentences with just one. Something like this:

> After requesting account deletion you'll have 14 days to cancel your request.

**Comment by meleu** [comment]
@luchaos I think we should also mention what will happen with user's achievements (if they have created some).

They need answers for these questions:
1. Will my achievements get deleted too?
2. If not, will the authorship of my achievements be changed?

Just as a reminder, as stated in the Developer's Code of Conduct, once an achievement creator uploads an achievement, they are giving it over to the community. So deleting the achievements together with the user's account is **NOT** something we want.

**Comment by luchaos** [comment]
@meleu as none of that would happen (achievements getting deleted, changing authorship) I did not include any mentions of that. Having created anything on the platform, being it achievements, leaderboards, comments, etc, does not fall into personal data and thus will remain.

**Comment by televandalist** [comment]
I've been crying about Cheevo Adoption forever and it seems like it would get a nice test run here. 

1. Dev requests account deletion.
2. Disclaimer comes up telling them that doing so will result in loss of authorship of their achievements. 
3. Once account is deleted, authorship goes to RAdmin, or a similar account. 
4. Active devs can "adopt" these achievements and take proper responsibility for them as if they are their own. 

Additionally, this would allow for devs to "disown" sets they no longer have interest in maintaining and there would also be somewhere to send cheevos by banned devs. 

Obviously there would need to be some specific guidelines, regulations, and criteria involved, but I cannot imagine this not being a win/win situation. Abandoned achievements would have a home with devs interested in *officially* maintaining them. The ticket count would take a huge blow as well. Everybody wins. 

It could even evolve into my original, albeit more extreme, Cheevo Adoption suggestion that's buried on Discord somewhere ;)

Also, the placeholder account/adoption system could be a temporary solution until there's a proper contribution credit system in place. As long as a paper trail is kept, said contribution credits could be made retroactively.

I'm willing to keep track of everything if everyone gets on board with it... when I "officially" come back, of course ;)

**Comment by luchaos** [comment]
I'm not sure I follow :) There is no "loss of authorship" - nor is it planned?
What _should_ happen down the line is that the "single responsibility" comes to an end - where _anyone_ who feels like it can contribute to an achievements improvement and not only the original author is bothered with ticket requests.

**Comment by meleu** [comment]
I replied @televandalist in #520

**Comment by MrOwnership** [comment]
There are a few places on the site that will show as "Unknown" after a user account is deleted.

Forum Topic List and Forum Posts
image
image

Developer Authorship
image

Game/User Wall Comments
image


https://github.com/RetroAchievements/RAWeb/blob/83f81e38e1cb7dc73f5428f9747c0d85aef2e18b/lib/render/user.php#L21-L23
Updating to above code to return just the username rather than "Unknown" would at least give proper credit and authorship to the deleted user.
```
    if (!$userCardInfo) {
        return '<i>' . $user . '</i>';
    }
```
image
image
image
image

**Comment by luchaos** [comment]
This is not how account deletion usually works, and very much intentional.
Users, especially developers, have to aware that they will lose all their precious credit if they choose to have their account deleted. And will not be able to sign up with the same username again. Otherwise we open the door for impersonation and a come and go culture that I'm very much against.

Our server is in Europe, GDPR is a thing. This is serious - if a user wants their account to be deleted then this means to the public eye and in our database all personal data is gone. This includes the username. We might keep references internally to retain data integrity - but the username _must not_ be seen anywhere.

**Comment by rzumer** [comment]
Nice PR, but this feature should be accessible for banned users too ;)

**Review by meleu — APPROVED** [review]

**Pull request merged**